docs/features/techdocs/extensions.md

```diff
@@ -0,0 +1,62 @@
+---
+id: extensions
+title: Using TechDocs Extensions
+sidebar_label: Using TechDocs Extensions
+description: How to use the built-in TechDocs extension points
+---
+
+# TechDocs Backend Extensions
+
+The TechDocs backend plugin provides the following extension points:
+
+- `techdocsPreparerExtensionPoint`
+  - Register a custom docs [PreparerBase extension](https://backstage.io/docs/reference/plugin-techdocs-node.preparerbase/)
+  - Ideal for when you want a custom type of docs created for a specific entity type
+- `techdocsBuildsExtensionPoint`
+  - Allows overriding the build phase Winston log transport (by default does not log to console)
+  - Allows overriding the [DocsBuildStrategy](https://backstage.io/docs/reference/plugin-techdocs-node.docsbuildstrategy/)
+- `techdocsPublisherExtensionPoint`
+  - Register a custom docs publisher
+- `techdocsGeneratorExtensionPoint`
+  - Register a custom [TechdocsGenerator](https://backstage.io/docs/reference/plugin-techdocs-node.techdocsgenerator/)
+
+Extension points are exported from `@backstage/plugin-techdocs-backend`.
+
+## Examples
+
+### Log TechDocs Build phase details to console
+
+By default, the TechDocs build phase does not log out much information while building out TechDocs. However, the
```

Comment:
This statement is fairly confusing to me as there's an entire UI for this in the app that lets you view a pretty verbose output of the build process. Could we perhaps clarify this a bit more?

Reply:
Interesting, if that exists I completely missed it. I kept having the build fail and when I would click to open the output in the Backstage UI it would simply say that mkdocs exited with code 1. No other information that I could see. So I had to dig into node_modules packages and deliberately console log things to get the reason why mkdocs failed. Interestingly, there were logger calls being made but they were not being put into the console or into the UI that I could find. Maybe I missed something! Let me see if I can reproduce an error again.

Reply:
Just so we are on the same page, when you use the default build locally when it's building there's a button to see the logs in the UI which gets you this:

![Screenshot 2024-09-09 at 9 22 21 AM](https://github.com/user-attachments/assets/87790e9f-aabc-486f-b263-5eaaf5308696)

![Screenshot 2024-09-09 at 9 26 02 AM](https://github.com/user-attachments/assets/a528fc15-8931-4336-ac95-5f60b1b377be)

I can see it being helpful to have those logs show up in the backend for other tools though, just feel that the wording you used is confusing given this.

Reply:
Yes, I remember seeing that but I do not remember all of the logs within the build stage of the plugin showing up there. For example, it would show that mkdocs failed with exit code 1, but the exact output of the mkdocs process was not showing up in those logs, so you could not see the error that mkdocs was outputting IIRC.

Reply:
Okay, I reproduced the issue. It's been a bit since I fought this, but I am now seeing a little more clearly my original confusion, as the error output was this:

```
error: Failed to build the docs page for entity component:default/switchboard: Failed to generate docs from /var/folders/w8/1bs55y1x3p1c2l142p9sr4kjj05g4m/T/backstage-uVQVbw into /private/var/folders/w8/1bs55y1x3p1c2l142p9sr4kjj05g4m/T/techdocs-tmp-e13ipr; caused by unknown error 'Command mkdocs failed, exit code: 1' {"timestamp":"2024-09-10T14:44:56.669Z"}
```

<img width="1728" alt="Screenshot 2024-09-10 at 09 45 25" src="https://github.com/user-attachments/assets/dac1a598-0732-446a-9038-e4c03390c83c">

Now, I'm seeing that I completely missed the actual error which was way at the top, not near the mkdocs output. Since I didn't see that, I went down a giant rabbit hole that lead to this PR.

My original attempt was to get the output from the child mkdocs process, which is swallowed by the logger that is provided to the `runCommand`. This was found by manually going into the node_modules plugin-techdocs-node package where the following is found:

  ```
await runCommand({
            command: 'mkdocs',
            args: ['build', '-d', outputDir, '-v'],
            options: {
              cwd: inputDir,
            },
            logStream,
          });
```

I noticed that the logStream that was provided was doing nothing, it was simply a PassThrough and the PassThrough was not not showing its results anywhere!

So after spending several hours of manually debugging console.log outputs to try and figure out why I realized maybe I needed to provide a way to output that child processes result.

Part of that process involved me trying to figure out how to get the TechDocs mkdocs child process output into the console, which resulted in this PR.

That said, it appears that it turning on output of the TechDocs logs to the console doesn't actually output the child process logs anyway. So I guess I'll update this line as per your request.

Reply:
Yeah, that's no good, would you be open to logging an issue about this? Feel like there must be something we can do to make the error more obvious.

Going back to my comment, it was more just wanting to point out: "there are logs in the UI but there may be cases where you want to change that".

Reply:
@awanlin Oh no worries! I totally get what you are saying now. I was just trying to go down memory lane and figure out why the heck I made the PR in the first place :D I'll make a ticket.

Reply:
Issue created: https://github.com/backstage/backstage/issues/26608